src/App.vue

```diff
@@ -539,22 +536,23 @@ export default class App extends Vue {
 
   /** Opens Auth Web in a new tab with the survey query param. */
   protected launchFilingSurvey (): void {
-    const iaSurveyId = sessionStorage.getItem('IA_SURVEY_ID')
-
     // safety check
-    if (iaSurveyId) {
-      const url = `${sessionStorage.getItem('AUTH_WEB_URL')}?survey=${iaSurveyId}`
+    if (this.iaSurveyId > 0) {
+      const url = `${sessionStorage.getItem('AUTH_WEB_URL')}?survey=${this.iaSurveyId}`
       this.window.open(url, '_blank', 'noreferrer')
     }
   }
 
-  /** Called to save or delete "hide survey" cookie. */
-  protected updateSurveyCookie (doNotShow: boolean): void {
+  /** Called to update "do not show again" state. */
+  protected doNotShowSurvey (doNotShow: boolean): void {
     if (doNotShow) {
-      // hide survey for 30 days
-      Cookies.set(this.IA_SURVEY_KEY, 'do-not-show', { domain: this.domain, expires: 30 })
+      // safety check
```

Comment:
This method should not be called if IA Survey ID was 0 in the first place, but we still don't want to set a local storage variable with an invalid id.